### PR TITLE
BigQuery lib. Add support for customer supplied encryptiion keys

### PR DESCRIPTION
**Pull request by alexblyzn**
More details for this change in https://github.com/GoogleCloudPlatform/google-cloud-java/issues/2866

### CODE REVIEW

**Comment by alexblyzn** [comment]
I have integration tests for the change as well, but I don't have a good way to commit them. They require KMS key setup as dependency, it seems somewhat unreasonable to make everyone go through that setup if they want to run tests.

The tests:
https://github.com/alexblyzn/google-cloud-java/commit/7a34991f2c5b434e31d05dbd59e03a51ebdfcd77#diff-af63ea32c5525b76f3837bc5aa8e5a25

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/CopyJobConfiguration.java [hunk 1]
```diff
@@ -19,6 +19,7 @@
 import static com.google.common.base.Preconditions.checkNotNull;
 
 import com.google.api.services.bigquery.model.JobConfigurationTableCopy;
+import com.google.cloud.bigquery.LoadJobConfiguration.Builder;
```
Comment (pongad) [inline google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/CopyJobConfiguration.java]: How is this used? I find it a little strange that a copy configuration uses load config builder.

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/EncryptionConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2016 Google LLC
```
Comment (pongad) [inline google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/EncryptionConfiguration.java]: 2018?

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/EncryptionConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+ * limitations under the License.
+ */
+
+package com.google.cloud.bigquery;
+
+import com.google.common.base.MoreObjects;
+import java.io.Serializable;
+import java.util.Objects;
+
+public final class EncryptionConfiguration implements Serializable {
```
Comment (pongad) [inline google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/EncryptionConfiguration.java]: Consider using `@AutoValue` for this

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/QueryJobConfiguration.java [hunk 4]
```diff
@@ -24,6 +24,7 @@
 import com.google.cloud.bigquery.JobInfo.CreateDisposition;
 import com.google.cloud.bigquery.JobInfo.WriteDisposition;
 import com.google.cloud.bigquery.JobInfo.SchemaUpdateOption;
+import com.google.cloud.bigquery.LoadJobConfiguration.Builder;
```
Comment (pongad) [inline google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/QueryJobConfiguration.java]: same comment about Builder

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/CopyJobConfiguration.java (hunk 1 above)
Comment (alexblyzn) [reply google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/CopyJobConfiguration.java]: not used, removed.

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/EncryptionConfiguration.java (hunk 2 above)
Comment (alexblyzn) [reply google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/EncryptionConfiguration.java]: Updated

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/EncryptionConfiguration.java (hunk 3 above)
Comment (alexblyzn) [reply google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/EncryptionConfiguration.java]: It needs to be parsed and serialized to proto. It is easier for me to follow existing pattern here.

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/QueryJobConfiguration.java (hunk 4 above)
Comment (alexblyzn) [reply google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/QueryJobConfiguration.java]: Removed

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/EncryptionConfiguration.java (hunk 3 above)
Comment (pongad) [reply google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/EncryptionConfiguration.java]: OK with me. I suppose it's not hard to change later.

[the remark this answers is not in the document]
Comment (pongad) [reply google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/EncryptionConfiguration.java]: I think we should do this.

[the remark this answers is not in the document]
Comment (alexblyzn) [reply google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/EncryptionConfiguration.java]: Added.

**Comment by alexblyzn** [comment]
Is there anything else I need to change before you can accept the request?

**Pull request merged**